gltfTutorial/gltfTutorial_014_AdvancedMaterial.md

```diff
@@ -0,0 +1,450 @@
+
+```javascript
+"meshes" : {
+  "mesh0" : {
+    "primitives" : [ {
+      "attributes" : {
+        "POSITION" : "positionsAccessor",
+        "NORMAL" : "normalsAccessor",
+        ...
+      },
+      ...
+    } ]
+  }
+},
+...
+"techniques": {
+  "advancedTechnique": {
+    ...
+    "attributes": {
+      "a_position": "positionParameter",
+      "a_normal": "normalParameter"
+    },
+    ...
+    "parameters": {
+      "positionParameter" : {
+        "type": 35665,
+        "semantic": "POSITION"
+      },
+      "normalParameter" : {
+        "type": 35665,
+        "semantic": "NORMAL"
+      },
+      ...
+    }
+  },
+},
+"programs": {
+  "advancedProgram": {
+    ...
+    "attributes": [
+      "a_position",
+      "a_normal"
+    ]
+  }
+},
+"accessors" : {
+  ...
+  "positionsAccessor" : {
+    ...
+  },
+  "normalsAccessor" : {
+    ...
+  },
+  ...
+},
+```
+
+Summarizing from the previous sections:
+
+- The `mesh` contains [mesh primitive attributes](gltfTutorial_008_Meshes.md#mesh-primitive-attributes). One of them is the new `NORMAL` attribute. It refers to the [accessor](gltfTutorial_005_BuffersBufferViewsAccessors.md#accessors), which provides the data of the vertex normals.
+- The `program` contains a list of all `attribute` variables that are contained in its vertex shader, including the `a_normal` attribute.
+- The `technique` contains the `attributes` dictionary whose entry for the `a_normal` attribute points to the `technique.parameters` dictionary, which defines the properties of this attribute, as shown in the section about [technique parameter values](gltfTutorial_013_MaterialsTechniques.md#technique-parameter-values).
+
+
+
+## The new `uniform` parameters for the matrices
+
+The vertex shader contains several `uniform` variables. One of them is the `u_normalMatrix`, which is the matrix that is used for transforming the normals when the object is transformed with the model-view matrix:
+
+```glsl
+uniform mat3 u_normalMatrix;
+...
+```
+
+Like the other uniforms, this uniform is listed in the `uniforms` dictionary of the `technique` and is further defined in the `technique.parameters`:
+
+```javascript
+"techniques": {
+  "advancedTechnique": {
+    ...
+    "uniforms": {
+      ...
+      "u_normalMatrix": "normalMatrixParameter",
+      ...
+    },
+    "parameters": {
+      ...
+      "normalMatrixParameter": {
+        "type": 35675,
+        "semantic": "MODELVIEWINVERSETRANSPOSE"
+      },
+    }
+  }
+},
+```
+
+In contrast to the other uniforms that have been added to the advanced material, this uniform has a `semantic` property, with the value `"MODELVIEWINVERSETRANSPOSE"`. Also note that its `type ` is `35675`, which stands for `GL_FLOAT_MAT3`, and thus indicates a 3&times;3 matrix (and not a 4&times;4 matrix). Particularly, this matrix consists of the upper left 3&times;3 entries of the model-view-matrix, transposed and inverted.

```

Comment:
I wasn't totally confident in my edits to the last sentence: "Particularly, this matrix..." Is the meaning correct here?